go/vt/vttablet/tabletserver/querythrottler/query_throttler.go

```diff
@@ -84,27 +84,28 @@ func (qt *QueryThrottler) Shutdown() {
 
 // Throttle checks if the tablet is under heavy load
 // and enforces throttling by rejecting the incoming request if necessary.
-// Note: This method performs lock-free reads of config and strategy for optimal performance.
-// Config updates are rare (default: every 1 minute) compared to query frequency,
-// so the tiny risk of reading slightly stale data during config updates is acceptable
-// for the significant performance improvement of avoiding mutex contention.
+// This method uses RWMutex read locks to safely access qt.cfg and qt.strategy
+// while allowing concurrent reads from multiple goroutines.
 func (qt *QueryThrottler) Throttle(ctx context.Context, tabletType topodatapb.TabletType, parsedQuery *sqlparser.ParsedQuery, transactionID int64, options *querypb.ExecuteOptions) error {
-	// Lock-free read: for maximum performance in the hot path as cfg and strategy are updated rarely (default once per minute).
-	// They are word-sized and safe for atomic reads; stale data for one query is acceptable and avoids mutex contention in the hot path.
-	if !qt.cfg.Enabled {
+	qt.mu.RLock()
+	cfg := qt.cfg
+	strategy := qt.strategy
```

Comment:
If this is enough to avoid the race, then I wonder if we can make the strategy an atomic pointer instead?